v2/cacheutils/cache.go

```diff
@@ -12,7 +13,7 @@ import (
 	"strings"
 
 	"github.com/golang/glog"
-	api "github.com/kubeflow/pipelines/v2/third_party/kfp_api"
+	api "github.com/kubeflow/pipelines/v2/kfp_api"
```

Comment:
None blocker, kfp_api does not conform to go package name style. _ is not allowed.

Also, why is kfp_api here? I assume it should be in KFP/backend/api/go_client. Did I miss anything?

Reply:
I just copy the generated protocol buffer code to this package. 
After your PR (which refactors the pipeline_spec package) gets merged, we can remove this package